### PR TITLE
Setup depedabot with auto-merge

### PR DESCRIPTION
**Pull request by janisz**
This PR configures depedabot for this repo and auto-merge it's PRs if all checks passes.
This is needed as old dependencies in this projects cause failures on projects that depends on it when they try to update. 
- https://github.com/stackrox/stackrox/pull/6174#issuecomment-1567193736

Configuration is similar to one done in other stackrox repositories.

Tested [here](https://github.com/stackrox/helmtest/actions/runs/5123643334/jobs/9214446347?pr=28#step:3:11) but PR needs to be created by dependabot :(

### CODE REVIEW

File: .github/workflows/auto-merge.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  pull_request_target:
+
+jobs:
+  auto-merge:
+    runs-on: ubuntu-latest
+    if: github.actor == 'dependabot[bot]'
+    steps:
+    - uses: ahmadnassri/action-dependabot-auto-merge@v2.6
+      with:
+        github-token: '${{ secrets.ROBOT_ROX_GITHUB_TOKEN }}'
```
Comment (janisz) [inline .github/workflows/auto-merge.yml:13]: @SimonBaeumer since I have no rights to this repo could you add this secret?

Comment (GrimmiMeloni) [reply .github/workflows/auto-merge.yml:13]: Added you @janisz (and Draco and Automation) as Admins.

Comment (janisz) [reply .github/workflows/auto-merge.yml:13]: I've added secret

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    reviewers:
+      - "stackrox/backend-dep-updaters"
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: 'weekly'
+      day: 'wednesday'
+    open-pull-requests-limit: 1
+    reviewers:
+    - "stackrox/backend-dep-updaters"
```
Comment (dhaus67) [inline .github/dependabot.yml:22]: Screenshot 2023-05-31 at 05 48 09

**Review by dhaus67 — APPROVED** [review]
LGTM, one nit regarding the newline.

I suppose we should wait for one approval from Draco to check whether they are OK with the CODEOWNERS change.

**Review by vladbologa — APPROVED** [review]

**Pull request merged**